crates/sui-json-rpc/src/name_service.rs

```diff
@@ -85,11 +116,11 @@ impl NameServiceConfig {
         }
     }
 
-    pub fn record_field_id(&self, domain: &Domain) -> ObjectID {
+    pub fn record_field_id(&self, domain: &Domain, registry_id: Option<ObjectID>) -> ObjectID {
```

Comment:
Can you add a doc comment here explaining what the purpose is of supplying this `registry_id` field (given `self` already has its own `registry_id`)?

Reply:
Will indeed add a comment.
For context: You can resolve a `record_field_id` either on the main registry ID (which is ~static, part of self), or on the namespace ID (for a subdomain lookup).